Experiments/ExperimentsTests/VariationCacheTests.swift

```diff
@@ -0,0 +1,36 @@
+import XCTest
+@testable import Experiments
+
+final class VariationCacheTests: XCTestCase {
+    func test_variation_is_nil_when_the_value_does_not_exist() throws {
+        // Given
+        let userDefaults = try XCTUnwrap(UserDefaults(suiteName: UUID().uuidString))
+
+        // When
+        let cache = VariationCache(userDefaults: userDefaults)
+
+        // Then
+        XCTAssertNil(cache.variation(for: .applicationPasswordAuthentication))
+    }
+
+    func test_correct_variation_is_returned_after_setting_it() throws {
+        // Given
+        let userDefaults = try XCTUnwrap(UserDefaults(suiteName: UUID().uuidString))
+        let cache = VariationCache(userDefaults: userDefaults)
+
+        // When
+        try cache.assign(variation: .treatment, for: .applicationPasswordAuthentication)
```

Comment:
Nit: should we create mock tests instead to avoid having to update this test when the experiment is removed?

Reply:
Good point. Done in https://github.com/woocommerce/woocommerce-ios/pull/8806/commits/e0d9ee5a75b8da40669fe2e7d605238f50b1b73b